testdata/p4_16_samples_outputs/gauntlet_ctrl_plane_hdr.p4-stderr

```diff
@@ -0,0 +1,3 @@
+gauntlet_ctrl_plane_hdr.p4(25): [--Wwarn=invalid_header] warning: accessing a field of a potentially invalid header ctrl_hdr
```

Comment:
Making the warning function check the context itself would be the best solution in my opinion, but I am not sure how exactly it could be done. I agree about the annotation, I will check it for some of the subsequent commits.

Reply:
I have filed #2895 and I will try to do it myself.